File: kotlin/kotlin-vs-java/java-example/src/main/java/org/example/ex09/Exam09.java
package org.example.ex09;

/**
 * when
 */
public class Exam09 {
    
    public Exam09(StoreUser storeUser) {
        
        if (storeUser.getRole().equals("MASTER")) {
            
        } else if (storeUser.getRole().equals("ADMIN")) {
            
        } else if (storeUser.getRole().equals("USER")) {
            
        } else {
            
        }
        
        switch (storeUser.getRole()) {
            case "MASTER":
                break;
            case "ADMIN":
                break;
            case "USER":
                break;
            default:
        }
        
        try {
            
        } catch (Exception e) {
            if (e instanceof NullPointerException) {
                
            } else if (e instanceof NumberFormatException) {
                
            }
        }

    }

    public static void main(String[] args) {
        new Exam09(new StoreUser());
    }
}

class StoreUser {
    private String name;
    private String role;

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getRole() {
        return role;
    }

    public void setRole(String role) {
        this.role = role;
    }
}

File: kotlin/kotlin-vs-java/kotlin-example/src/main/kotlin/ex09/Exam09.kt
package ex09

import java.lang.NullPointerException
import java.lang.RuntimeException

/**
 * when
 */
fun main() {
    val storeUser = StoreUser()
    println(storeUser.role)

    val result: String = when(storeUser.role) {
        "MASTER" -> {
            "master"
        }
        "ADMIN" -> {
            "admin"
        }
        "USER" -> {
            "user"
        }
        else -> {
            "guest"
        }
    }

    println(result)

    var exception = RuntimeException()

    when(exception) {
        is RuntimeException -> {

        }
        is NullPointerException -> {

        }
    }

    var number = 10

    when(val n = number % 2) {
        0 -> {
            println("짝")
        }
        else -> {
            println("홀")
        }
    }

    val result2:Int = when {
        number % 2 == 0 -> {
            100
        }
        else -> {
            200
        }
    }

    println(result2)
}

data class StoreUser(
    var name: String?= "guest",
    var role: String?= "guest",
)